### PR TITLE
feat: add parameter to customize token lifespan

### PR DESCRIPTION
**Pull request by endroca**

### CODE REVIEW

**Comment by xgp** [comment]
Thanks @endroca . 2 things:
1. create an issue for tracking this
2. please run `mvn fmt:format` before committing, as the diff isn't readable because of your formatting changes

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticator.java [hunk 1]
```diff
@@ -143,6 +149,17 @@ private boolean isActionTokenPersistent(AuthenticationFlowContext context, boole
...
 
+  private OptionalInt getActionTokenLifeSpan(
+      AuthenticationFlowContext context, String defaultValue) {
+    String lifespan = get(context, ACTION_TOKEN_LIFE_SPAN, defaultValue);
+
+    if ("".equals(lifespan)) {
+      return OptionalInt.empty();
+    }
+
+    return OptionalInt.of(Integer.parseInt(lifespan));
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticator.java:160]: Should catch a `NumberFormatException` here, as the string doesn't get validated

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticator.java [hunk 2]
```diff
@@ -83,12 +85,14 @@ public void action(AuthenticationFlowContext context) {
...
       context
-              .getAuthenticationSession()
-              .setAuthNote(AbstractUsernameFormAuthenticator.ATTEMPTED_USERNAME, email);
+          .getEvent()
+          .detail(AbstractUsernameFormAuthenticator.ATTEMPTED_USERNAME, email)
+          .event(EventType.LOGIN_ERROR)
+          .error(Errors.INVALID_EMAIL);
+      context
+          .getAuthenticationSession()
+          .setAuthNote(AbstractUsernameFormAuthenticator.ATTEMPTED_USERNAME, email);
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticator.java:95]: Is there a change here? I can't tell.

**Review by xgp — CHANGES_REQUESTED** [review]
Couple of changes. Otherwise looking good.

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticator.java (hunk 1 above)
Comment (endroca) [reply src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticator.java:160]: True, I didn't validate that... there are two approaches here: Forcing an unknown error for Keycloak and preventing login, or going with the default settings and logging an error.

I'm more inclined to use the default configuration. What do you think?

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticator.java (hunk 2 above)
Comment (endroca) [reply src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticator.java:95]: I didn’t change that; it was probably the formatter.

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticatorFactory.java [hunk 3]
```diff
@@ -100,7 +100,15 @@ public List<ProviderConfigProperty> getConfigProperties() {
         "Toggle whether magic link should be persistent until expired.");
     actionTokenPersistent.setDefaultValue(true);
 
-    return List.of(createUser, updateProfile, updatePassword, actionTokenPersistent);
+    ProviderConfigProperty actionTokenLifeSpan = new ProviderConfigProperty();
+    actionTokenLifeSpan.setType(ProviderConfigProperty.TEXT_TYPE);
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticatorFactory.java:104]: Looks like I didn't submit this in my first review. This should be `STRING_TYPE`, not `TEXT_TYPE`, which is for a textarea.

**Review by xgp — CHANGES_REQUESTED** [review]

**Review by xgp — APPROVED** [review]

**Pull request merged**